lib/src/year_picker.dart

```diff
@@ -0,0 +1,591 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_localizations/flutter_localizations.dart';
+import 'package:intl/intl.dart' as intl;
+
+import 'date_picker_keys.dart';
+import 'day_type.dart';
+import 'i_selectable_picker.dart';
+import 'month_picker_selection.dart';
+import 'semantic_sorting.dart';
+import 'styles/date_picker_styles.dart';
+import 'styles/layout_settings.dart';
+import 'utils.dart';
+
+const Locale _defaultLocale = Locale('en', 'US');
+
+/// Year picker widget.
+class YearPicker<T extends Object> extends StatefulWidget {
+  YearPicker._({
+    Key? key,
+    required this.selectionLogic,
+    required this.selection,
+    required this.onChanged,
+    required this.firstDate,
+    required this.lastDate,
+    this.datePickerLayoutSettings = const DatePickerLayoutSettings(),
+    this.datePickerKeys,
+    required this.datePickerStyles,
+  })  : assert(!firstDate.isAfter(lastDate)),
+        assert(
+            selection.isEmpty || !selection.isBefore(firstDate),
+            'Selection must not be before first date. '
+            'Earliest selection is: ${selection.earliest}. '
+            'First date is: $firstDate'),
+        assert(
+            selection.isEmpty || !selection.isAfter(lastDate),
+            'Selection must not be after last date. '
+            'Latest selection is: ${selection.latest}. '
+            'First date is: $lastDate'),
+        super(key: key);
+
+  /// Creates a year picker where only one single year can be selected.
+  ///
+  /// See also:
+  /// * [YearPicker.multi] - year picker where many single years
+  ///   can be selected.
+  static YearPicker<DateTime> single(
+      {Key? key,
+      required DateTime selectedDate,
+      required ValueChanged<DateTime> onChanged,
+      required DateTime firstDate,
+      required DateTime lastDate,
+      DatePickerLayoutSettings datePickerLayoutSettings =
+          const DatePickerLayoutSettings(),
+      DatePickerStyles? datePickerStyles,
+      DatePickerKeys? datePickerKeys,
+      SelectableDayPredicate? selectableDayPredicate,
+      ValueChanged<DateTime>? onMonthChanged}) {
```

Comment:
What is this callback for? Is it possible to change month in this picker?

Reply:
You are right, it should be onYearChanged 😄